triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/ServiceTest.scala

```diff
@@ -219,6 +221,35 @@ class ServiceTest extends AsyncFlatSpec with Eventually with Matchers with Postg
     val testJdbcConfig = JdbcConfig(postgresDatabase.url, "operator", "password")
     assert(ServiceMain.initDatabase(testJdbcConfig).isRight)
     dropDatabase()
+    disconnectFromPostgresqlServer()
+    succeed
+  }
+
+  it should "add running triggers to the database" in {
```

Comment:
Instead of adding a bunch of separate tests for DB, you might want to consider abstracting over the backend in the tests and then instantiating all tests once with an in-memory backend and once with a SQL backend. You can still have additional tests for each backend if required but at least you test that the basic functionality works correctly with both.

Reply:
Yes good point, I thought of this too. It has the potential downside of making the tests too slow, but would be much easier to make the tests comprehensive. I'll explore this in a later PR.